pkg/server/plugin/datastore/datastore.go

```diff
@@ -93,30 +91,28 @@ type BySelectors struct {
 	Match     MatchBehavior
 }
 
-type GetNodeSelectorsRequest struct {
-	SpiffeId string //nolint: golint
-	// When enabled, read-only connection will be used to connect to database read instances. Some staleness of data will be observed.
-	TolerateStale bool
+type JoinToken struct {
+	Token  string
+	Expiry time.Time
 }
 
-type GetNodeSelectorsResponse struct {
-	Selectors *NodeSelectors
+type NodeSelectors struct {
+	SpiffeID  string
+	Selectors []*common.Selector
 }
 
-type JoinToken struct {
-	// Token value
-	Token string
-	// Expiration in seconds since unix epoch
-	Expiry time.Time
+type Pagination struct {
+	Token    string
+	PageSize int32
 }
 
 type ListAttestedNodesRequest struct {
-	ByExpiresBefore   *wrapperspb.Int64Value
-	Pagination        *Pagination
 	ByAttestationType string
+	ByBanned          *bool
+	ByExpiresBefore   *time.Time
```

Comment:
Maybe we can just use `time.Time` here and detect in the datastore if it is set to the zero value before applying?

```
if !req.ByExpiresBefore.IsZero() {
    // apply filter
}
```

Reply:
Good point, a zero time is effectively the same as don't filter